sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/v2Commands.scala

```diff
@@ -317,6 +331,24 @@ case class WriteDelta(
 
   override lazy val references: AttributeSet = query.outputSet
 
+  lazy val inputRowAttrs: Seq[Attribute] = {
```

Comment:
The biggest change here is that we use column ordinals from projections to find actual input attributes instead of relying on a fake schema we expected but did not enforce/validate before.